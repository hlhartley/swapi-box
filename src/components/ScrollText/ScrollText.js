import React from 'react';
import '../../main.scss';

const ScrollText = (props) => {
    return (
        <div className='scroll-text-container'>
        <div className="fade"></div>
        <section className="star-wars">
        <div className="crawl">
            <div className="title">
            <p>Episode IV </p>
            <h1>{props.films.title}</h1>
            </div>
            <p className='scroll-text'>{props.films.scrollText}</p>
            <p>- released: {props.films.release}</p>
        </div>
        </section>
        </div>
    )
}

export default ScrollText;