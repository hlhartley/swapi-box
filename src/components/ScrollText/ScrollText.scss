.scroll-text {
    background-color: #f2f2f2;
    color: blue;
    padding: 40px;
}