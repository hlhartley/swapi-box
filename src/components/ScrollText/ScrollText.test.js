import React from 'react';
import ReactDOM from 'react-dom';
import ScrollText from './ScrollText';
import { shallow } from 'enzyme';

let wrapper;
describe('ScrollText', () => {

  const film = { 
    episode: 4,
    scrollText: "It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secreâ€¦",
    title: "A New Hope"
  }

  beforeEach(() => {
    wrapper = shallow(<ScrollText film={film}/>); 
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.unmountComponentAtNode(div);
})

  it('should exist', () => {   
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
