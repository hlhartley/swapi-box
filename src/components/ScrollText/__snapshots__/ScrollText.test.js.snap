// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText should match the snapshot with all data passed in correctly 1`] = `
<div
  className="scroll-text-container"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          Episode IV 
        </p>
        <h1>
          A New Hope
        </h1>
      </div>
      <p
        className="scroll-text"
      >
        It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secreâ€¦
      </p>
      <p>
        - released: 
      </p>
    </div>
  </section>
</div>
`;
